tests/integration/corrupt-dump-fuzzer.tcl

```diff
@@ -147,10 +147,21 @@ foreach sanitize_dump {no yes} {
                         if {$dbsize != [r dbsize]} {
                             puts "unexpected keys"
                             puts "keys: [r keys *]"
-                            puts $sent
+                            puts "commands leading to it:"
+                            foreach cmd $sent {
+                                foreach arg $cmd {
+                                    puts -nonewline "[string2printable $arg] "
+                                }
+                                puts ""
+                            }
                             exit 1
                         }
                     } err ] } {
+                        set err [format "%s" $err] ;# convert to string for pattern matching
+                        if {[string match "*SIGTERM*" $err]} {
+                            puts "payload that caused test to hang: $printable_dump"
+                            exit 1
+                        }
```

Comment:
should this block put into `if {$by_signal != 0 || $sanitize_dump == yes}`?
when `sanitize_dump == no` and crash by assert, we shouldn't report it.

Reply:
https://github.com/sundb/redis/actions/runs/6480479444/job/17596083922

```
HMGET _hash 315 
payload that caused test to hang: "\x10\x15\x15\x00\x00\x00\x06\x00\x00\x01\xAD\x01\x01\x01\x82\x5F\x31\x03\x02\x01\x02\x01\xFF\x0B\x00\xFD\xFE\xB7\x89\xFB\xCD\x76\xCD"
```

reproduce command:
```
debug set-skip-checksum-validation 1
config set sanitize-dump-payload no
restore _hash 0 "\x10\x15\x15\x00\x00\x00\x06\x00\x00\x01\xAD\x01\x01\x01\x82\x5F\x31\x03\x02\x01\x02\x01\xFF\x0B\x00\xFD\xFE\xB7\x89\xFB\xCD\x76\xCD" replace
HMGET _hash 315
```

Reply:
the SIGTERM is not terminating redis, it's terminating the test suite (TCL).
and that happens when there's a timeout (the test didn't complete in time), see #8837.
the reason it started happening now (nearly every day), is probably because of #12623.
what seems to happen is that an assertion happens, and then we try to collect the stack traces from the threads, and for some reason we hang (never get to print the stack trace), then the test suite terminates the test (hence SIGTERM).
so far we weren't able to figure out why it hangs.